shared/events/edgeware/types.ts

```diff
@@ -0,0 +1,149 @@
+import { Header, EventRecord } from '@polkadot/types/interfaces';
+
+/**
+ * Substrate lacks a block type that includes events as well, so we synthesize a type
+ * from the combination of headers and events.
+ */
+export interface SubstrateBlock {
+  header: Header;
+  events: EventRecord[];
+}
+
+/**
+ * To implement a new form of event, add it to this enum, and add its
+ * JSON interface below (ensure it is stringify-able and then parse-able).
+ */
+
+/** Special types for formatting/labeling purposes */
+export type SubstrateBalanceString = string;
+export type SubstrateBlockNumber = number;
+export type SubstrateAccountId = string;
+
+export interface ISubstrateSlash {
```

Comment:
seems we can use these as the enums? Is it possible even though they're interfaces?

```
switch (thing) {
  case ISubstrateSlash.kind:
    ...
}
```

Reply:
I don't think so, you can't use a type (interfaces are types) as a value. But I could swap out all the "kind" strings with a string enum.